### ISSUE TITLE
date.exe -u argument doesn't work.

### ISSUE THREAD

**Issue by hellishvictor** [uutils/coreutils#6396]
Hi, I think I've found a bug using the **-u** argument with date.exe. When is used with the _GnuWin32_ version it shows:
`date.exe -u -d "@1715372666" +"%F, %T %Z%z"`
**2024-05-10, 20:24:26 UTC+0000**

And when is not used:
`date.exe -d "@1715372666" +"%F, %T %Z%z"`
**2024-05-10, 21:24:26 Hora de verano GMT+0100**

But with _coreutils_, with or without the **-u**, it always shows the same time:
`coreutils date -d "@1715372666" +"%F, %T %Z%z"`
**2024-05-10, 20:24:26 +01:00+0100**
Here without the -u should be one hour more (21:24:26), and when used it should show +0000.

Cheers.

**Comment by DaveKram** [comment]
I decided to take a look at this - it seems like the following logic needs to be updated:

https://github.com/uutils/coreutils/blob/8cac375dddb4d4e87519c137c45fa831ac7f2619/src/uu/date/src/date.rs#L214-L246

The math up to this point is still using Local::now() (thus current_time still being a Local::now()). I'm thinking that the return value down below should use the `now` variable's offset() (since that takes into consideration UTC). I.E something like this:

`
let iter = std::iter::once(Ok(date.with_timezone(now.offset())));
`

Which changes the string representation of the date to be offset by `now`, which is based off of whether the -u UTC flag is set or not.

Would like someone to check this (since is my first look at the repo) - can make a PR if is accurate.

**Comment by CausingBrick** [comment]
I think this was fixed in #6503

Test as bellow after that meger:

```bash
# GNU date
➜ date -u -d "@1715372666" +"%F, %T %Z%z"
2024-05-10, 20:24:26 UTC+0000
➜ date -d "@1715372666" +"%F, %T %Z%z"
2024-05-11, 04:24:26 CST+0800

# uu_date
➜ cargo r -p uu_date -- -u -d "@1715372666" +"%F, %T %Z%z"
2024-05-10, 20:24:26 +00:00+0000
➜ cargo r -p uu_date -- -d "@1715372666" +"%F, %T %Z%z"
2024-05-11, 04:24:26 +08:00+0800
```

**Comment by hellishvictor** [comment]
Fixed, thank you!
